src/App.js

```diff
@@ -27,14 +37,49 @@ const MemeContainer = styled(Flex)`
   position: relative;
 `;
 
+const DEFAULT_IMG_URL = '';
+const HIDDEN_IMG_URL_INPUT = false;
+
+const bypassCorsUrl = url => `https://meme.st6.io/meme-image?imageUrl=${url}`;
+
 const App = () => {
   const isMobile = useMedia({ maxWidth: '40em' });
   const [theme, setTheme] = useState('light');
+  const [showUrlInput, setShowUrlInput] = useState(HIDDEN_IMG_URL_INPUT);
   const [topLabel, setTopLabel] = useState('Do the most meaningful meme...');
   const [bottomLabel, setBottomLabel] = useState('...of your life');
+  const [imageUrl, setImageUrl] = useState(DEFAULT_IMG_URL);
   const [imageSrc, setImageSrc] = useState(getRandomMeme());
   const ref = useRef(null);
 
+  const hideUrlInput = useCallback(() => {
+    setImageUrl(DEFAULT_IMG_URL);
+    setShowUrlInput(HIDDEN_IMG_URL_INPUT);
+  }, [setImageUrl, setShowUrlInput]);
```

Comment:
This looks strange. Your dependencies are setter functions. Definitely something is not OK here.

Reply:
`useCallback` requires a set of dependencies to be passed, I guess I could use an empty array here.